### PR TITLE
feat: Use cube-hierarchy-query to fetch hierarchies

### LINKED ISSUE #587: Hierarchies are not implemented according to the spec
**Describe the bug**
Hierarchy via `sh:inversePath` is broken and does not display any Hierarchies.
For the spec see https://cube.link/#hierarchies and especially the example 15

```ttl
PREFIX meta: <https://cube.link/meta/>
PREFIX sh: <http://www.w3.org/ns/shacl#>

meta:inHierarchy [
  a meta:Hierarchy ;
  meta:hierarchyRoot <https://ld.admin.ch/country/CHE> ;
  schema:name "CH - Canton" ;
  meta:nextInHierachy [
    schema:name "Canton" ;
    sh:path  [ sh:inversePath <http://schema.org/containedInPlace> ] ;
    sh:targetClass <https://schema.ld.admin.ch/Canton> ;
  ]
]
```

**To Reproduce**
1. Go to https://test.visualize.admin.ch/v/MaX3Y1_bjDBW
2. Click on 'copy and edit'
3. Click on 'color'
4. See no hierarchies appear

**Expected behavior**
Show the implemented hierarchies
based on `sh:inversePath` https://s.zazuko.com/66owwi
```ttl
_:genid-f965657b85144e86bd3f0e246dc124a0-F484FC645475EC192D1D0203B55924D1 a meta:Hierarchy ;
   meta:hierarchyRoot <https://ld.admin.ch/cube/dimension/warenkorbzoll/jq1> , <https://ld.admin.ch/cube/dimension/warenkorbzoll/jq11c> , <https://ld.admin.ch/cube/dimension/warenkorbzoll/jq11nc> ;
   meta:nextInHierarchy [
      sh:path [ sh:inversePath schema:isPartOf ]
   ] .
```

**Screenshots**
missing hierarchies
image

**Environment (please complete the following information):**
 - Visualize environment and version: TEST v3.7.5 ([906ca39](https://github.com/visualize-admin/visualization-tool/commit/906ca399f5e655365d01cc16e2d9cd5e8d59248e))
 - Browser and version Chrome 101

**Additional context**
NB: do you reuse `@zazuko/cube-hierarchy-query` at all or are you reinventing an implementation?

cc @ortnever @zellersabine @pierredumas

### PR DESCRIPTION
**Pull request by ptbrowne**
Using the new version of cube-hierarchy-query instead of custom code allows for
retrieval of the complete hierarchy for display in the right side filters.

- [x] ⚠️ Need to wait for @tpluscode to publish the new version (for now the code has been tested with a non published version of the library based on https://github.com/zazuko/cube-hierarchy-query/pull/9)

fix #587 #588

### CODE REVIEW

File: app/rdf/query-hierarchies.ts [hunk 1]
```diff
@@ -26,66 +24,80 @@ const queryDimensionValues = async (dimension: string) => {
...
+  if (cubeResults.length === 0) {
+    throw new Error("Could not find cube");
+  }
+  const cubeIri = cubeResults[0].cube.value;
+  const cube = await source.cube(cubeIri);
+  if (!cube) {
+    throw new Error("Could not find cube");
+  }
+  const hierarchy = cube.ptr
+    .any()
```
Comment (bprusinowski) [inline app/rdf/query-hierarchies.ts:81]: What does this "any" do?

**Review by bprusinowski — APPROVED** [review]
💯

File: app/rdf/query-hierarchies.ts (hunk 1 above)
Comment (ptbrowne) [reply app/rdf/query-hierarchies.ts:81]: Good question, I had not checked the code. I saw this as a way to reset the current pointer to be the whole graph.

The official documentation explains better

> Removes current pointers from the context and return an "any pointer". The returned object can be used to find any nodes in the dataset

https://zazuko.github.io/clownface/#/api?id=clownface%20any

Comment (bprusinowski) [reply app/rdf/query-hierarchies.ts:81]: Thanks 🙇

File: app/graphql/queries/data-cubes.graphql [hunk 2]
```diff
@@ -273,6 +273,12 @@ query DimensionHierarchy(
         ...hierarchyValueFields
         children {
           ...hierarchyValueFields
+          children {
+            ...hierarchyValueFields
+            children {
+              ...hierarchyValueFields
+            }
+          }
```
[the remark this answers is not in the document]
Comment (ptbrowne) [reply app/graphql/queries/data-cubes.graphql:281]: For now I block the depth of the tree to 4 levels. I think to be generic and allow any level we'd have to send a raw graphql query without passing through graphql codegen, and we'd lose the types.

**Comment by Rdataflow** [comment]
great, now we can fetch all the hierarchies meeting the spec 🚀

**Pull request merged**